pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -180,8 +180,11 @@ func NewController(client kubernetes.Interface, options Options) *Controller {
 	svcInformer := sharedInformers.Core().V1().Services().Informer()
 	out.services = out.createCacheHandler(svcInformer, "Services")
 
-	epInformer := sharedInformers.Core().V1().Endpoints().Informer()
-	out.endpoints = out.createEDSCacheHandler(epInformer, "Endpoints")
+	if features.EnableEndpointSliceController {
+		out.endpoints = newEndpointSliceController(out, sharedInformers)
+	} else {
+		out.endpoints = newEndpointsController(out, sharedInformers)
+	}
 
 	nodeInformer := sharedInformers.Core().V1().Nodes().Informer()
 	out.nodes = out.createCacheHandler(nodeInformer, "Nodes")
```

Comment:
move these two lines into else branch.

Reply:
We donot need  watching nodes anymore

Reply:
See comment above, can we do this in a follow up? We will still need nodes for hybrid endpoints+endpointslice case

Reply:
agree